implementations/micrometer-registry-statsd/src/test/java/io/micrometer/statsd/StatsdMeterRegistryPublishTest.java

```diff
@@ -0,0 +1,207 @@
+ * limitations under the License.
+ */
+package io.micrometer.statsd;
+
+import io.micrometer.core.Issue;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.Counter;
+import io.netty.handler.logging.LogLevel;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import reactor.core.publisher.Flux;
+import reactor.netty.DisposableChannel;
+import reactor.netty.tcp.TcpServer;
+import reactor.netty.udp.UdpServer;
+
+import java.time.Duration;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+/**
+ * Tests {@link StatsdMeterRegistry} metrics publishing functionality.
+ */
+class StatsdMeterRegistryPublishTest {
+
+    StatsdMeterRegistry meterRegistry;
+
+    @ParameterizedTest
+    @EnumSource(StatsdProtocol.class)
+    void receiveMetricsSuccessfully(StatsdProtocol protocol) throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(3);
+        DisposableChannel server = startServer(protocol, latch, 0);
+
+        final int port = server.address().getPort();
+
+        meterRegistry = new StatsdMeterRegistry(getUnbufferedConfig(protocol, port), Clock.SYSTEM);
+        startRegistryAndWaitForClient();
+        Counter counter = Counter.builder("my.counter").register(meterRegistry);
+        counter.increment();
+        counter.increment();
+        counter.increment();
+        assertThat(latch.await(3, TimeUnit.SECONDS)).isTrue();
+        meterRegistry.close();
+
+        server.disposeNow();
+    }
+
+    @ParameterizedTest
+    @EnumSource(StatsdProtocol.class)
+    void resumeSendingMetrics_whenServerIntermittentlyFails(StatsdProtocol protocol) throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(1);
+        DisposableChannel server = startServer(protocol, latch, 0);
+
+        final int port = server.address().getPort();
+
+        meterRegistry = new StatsdMeterRegistry(getUnbufferedConfig(protocol, port), Clock.SYSTEM);
+        startRegistryAndWaitForClient();
+        Counter counter = Counter.builder("my.counter").register(meterRegistry);
+        counter.increment(1);
+        assertThat(latch.await(1, TimeUnit.SECONDS)).isTrue();
+        // stop server
+        server.disposeNow();
+        // client will try to send but server is down
+        IntStream.range(2, 5).forEach(counter::increment);
+        // TODO wait until processor queue is drained...
+        // start server
+        latch = new CountDownLatch(3);
+        server = startServer(protocol, latch, port);
+        counter.increment(5);
+        counter.increment(6);
+        counter.increment(7);
+        assertThat(latch.await(1, TimeUnit.SECONDS)).isTrue();
```

Comment:
This fails when executing with the TCP server. It seems when we recreate a `TcpServer` on line 86, it is actually using the previously created `TcpServer`, which has a reference to the previous `CountDownLatch`, not the new one.

Reply:
Can you try 0.9.3.RELEASE. It might be related to this fix https://github.com/reactor/reactor-netty/commit/360ab0fd086a63d9bad235b1c399cc2051567f6f

Reply:
The same issue remained until I moved the `CountDownLatch` to a class field instead of a method parameter. I don't understand why the issue happens, but I've been able to work around it.